File: debug_excel.py
import pandas as pd

# Load Excel file and check what PO numbers exist for our test vendors
df = pd.read_excel('Service Agreement Table.xlsx', sheet_name='Service Agreements')

print("Checking PO numbers for our test vendors:")
print("="*60)

test_vendors = ['Mid South', 'Budd Group', 'John Bouchard']

for vendor in test_vendors:
    print(f"\nSearching for vendor containing: '{vendor}'")
    
    # Find vendors that contain our search term
    matches = df[df['Vendor'].str.contains(vendor, case=False, na=False)]
    
    if len(matches) > 0:
        for idx, row in matches.iterrows():
            print(f"  Found: '{row['Vendor']}'")
            print(f"    Admin: {row['Admin']}")
            print(f"    Manager: {row['Main Contact']}")
            print(f"    Current PO: {row['Current PO']}")
            print(f"    PO Start: {row['PO Start']}")
            print(f"    PO End: {row['PO End']}")
    else:
        print(f"  No matches found for '{vendor}'")

print("\n" + "="*60)
print("Test file PO numbers extracted:")
print("  12628 Mid South P26003063.pdf → P26003063")
print("  230006 The Budd Group P26000686.pdf → P26000686") 
print("  25-23487 John Bouchard P25063542.pdf → P25063542")

File: debug-excel.js
const XLSX = require('xlsx');

console.log('Debugging Excel data...');
const workbook = XLSX.readFile('Service Agreement Table (Rolling).xlsx');
const worksheet = workbook.Sheets['Service Agreements'];
const jsonData = XLSX.utils.sheet_to_json(worksheet, { header: 1 });

const headers = jsonData[0];
console.log('Looking for rate-related columns...');
headers.forEach((header, index) => {
  if (header && (header.toString().toLowerCase().includes('rate') || 
                 header.toString().toLowerCase().includes('amount') || 
                 header.toString().toLowerCase().includes('price') ||
                 header.toString().toLowerCase().includes('cost'))) {
    console.log(`Index ${index}: ${header}`);
  }
});

// Look for Mid South row and show all data
console.log('\nSearching for Mid South data...');
for (let i = 1; i < jsonData.length; i++) {
  const row = jsonData[i];
  if (row[0] && row[0].toString().toLowerCase().includes('mid south')) {
    console.log('\nMid South found at row:', i + 1);
    console.log('Vendor name:', row[0]);
    
    // Check for the rate you mentioned
    row.forEach((cell, index) => {
      if (cell && cell.toString().includes('10,942')) {
        console.log(`FOUND RATE - Column ${index} (${headers[index]}): ${cell}`);
      }
    });
    
    // Also check for weekly rates
    row.forEach((cell, index) => {
      if (cell && cell.toString().toLowerCase().includes('weekly')) {
        console.log(`WEEKLY FOUND - Column ${index} (${headers[index]}): ${cell}`);
      }
    });
    
    // Show scope or description columns that might contain rate info
    console.log('\nScope/Description columns:');
    if (row[1]) console.log(`Column 1 (${headers[1]}): ${row[1]}`);
    if (row[2]) console.log(`Column 2 (${headers[2]}): ${row[2]}`);
    
    break;
  }
}